online-shop/src/app/services/cart.service.ts

```diff
@@ -1,32 +1,38 @@
-import { Injectable } from '@angular/core';
+import {Injectable} from '@angular/core';
 
+import {Product} from '../models/product';
 
 @Injectable({
   providedIn: 'root'
 })
 
 export class CartService {
 
-  items = [];
+  items: Product[] = [];
 
-  constructor() { }
+  constructor() {
+  }
 
-  getCartContent(){
+  getCartContent() {
     return this.items;
   }
 
-  addToCart(product){
+  addToCart(product) {
     this.items.push(product);
-    
-  }
-  
-  removeFromCart(product){
-    this.items.splice(this.items.indexOf(product),1); //finds the index of the given parameter and deletes one element starting from that index 
-    return this.items;
   }
 
-  clearCart(){
+  clearCart() {
     this.items = [];
     return this.items;
   }
+
+  getTotal() {
+    let total = 0;
+
+    // tslint:disable-next-line:forin
+    for (const i in this.items) {
```

Comment:
Small hint: it is better to use for ... of instead of for ... in --> for ... in loops over the properties (including inherited properties) of an object, for ... of loops over iterables (e.g. arrays).